### ISSUE TITLE
Improvements to the "create a release plan" experience to ensure process adherence, automate decision-making, and enhance data integrity 

### ISSUE THREAD

**Issue by ladonnaq** [Azure/azure-sdk-tools#6262]
This requirement is a pre-requisite to automating attestation & approval of the APEX S360/Cloud Lifecycle KPIs.  Also, important to improving data integrity (gaps in data, validating correctness) and process adherence.  

Related epic to address actions for completing APEX requirement and all release plans (not just APEX)  - https://github.com/Azure/azure-sdk-tools/issues/9080

**Requirement:**  

_Improve user experience, release plan metadata, and process adherence:_ 

- The MVP release plan feature was designed for the creating release plans for APEX release scenarios, not non-APEX release scenarios. Brownfield Services find the UI to be confusing. The UI ask for the "Product's Lifecycle" but does not ask for what they are releasing (new API version only, new Beta SDK versions, new Stable SDK versions). This is causing confusing and leading teams to create incorrect release plans, which means they are not following the correct process & requirements for their scenario. 
- For example, the Azure Container Service  (ACS) Service and Offering are Brownfield and have existing REST APIs & SDKs for management plane only. They are adding new initial REST APIs & SDKs for data plane, so this scenario is akin to Greenfield and they are required to follow the Greenfield Service & Product process & requirements. Because the Azure Container Service offering (Product) is GA, they selected the Product's target lifecycle phase as GA. Today that means they created a GA release plan. Mike alerted us when the ACS team showed up to review an initial stable API with the Stewardship Board. 

_Automate tasks, decision-making, & adherence to process for specific scenario:_ 

- Enhancing the release plan workflows to automate tasks and streamline the user experience, the Release Planner app needs to know the release scenario. The release plan metadata needs to contain the relevant data to implement rules to toggle UI controls, automate relevant tasks, automate decision-making & adherence to process, etc. 
- For example, in September 2024 around 35 hours by Service partners & the SDK APEX PM manually determining the state of the APEX requirements and manually updating Azure Devops & Service 360/Cloud Lifecycle 

- [x] UI updates to Create a release plan feature.  Ready for review, info below screenshots.  (La Donna) 
- [x]  Update UI to include link to Info Meeting docs if the Info Meeting is required and is not completed  - https://github.com/Azure/azure-sdk-tools/issues/7237 & https://github.com/Azure/azure-sdk-tools/issues/9041
- [x] Workflow for create a release plan.  Diagram is under task list.  (La Donna)
- [x] _Data automation/back-end:_  Dependency on the requirement to update onboard feature to automatically identify APEX scenarios and update field value for isAPEX in the epic type = product work item. - https://github.com/Azure/azure-sdk-tools/issues/7084
- [x] Add picklist custom ADO field to the release plan work item. ADO field name = " isAPEX" with string picklist values of "yes" or "no".  (La Donna) 
- [x] Release Plan type picklist values simplified: Private Preview is now "API Spec - no SDKs", Public Preview is now "Beta SDKs", and GA is now "Stable SDKs". Added to existing picklist values, old values will remain until all release plan work items have been manually updated in ADO.  (La Donna)
- [ ] Add rules/logic to automate decision making and update data model to improve data integrity. (Engineering) The APEXStateFields, APEX launch criteria field, nonAPEXrelease new tabs have been updated with rules to populate APEX data related fields - https://microsoft-my.sharepoint.com/:x:/p/ladonnaquinn/ETD9YCt4cfxHlL4mC1KlVygBIQss1mm6NMIpu3_BdOJhJQ?e=94BLHb&nav=MTVfe0I0REYxMEVCLUZDOEUtNDlGMi05QTEyLTIyMTAxNTUyQzJCM30  
- [ ] After the automation is complete: export release plan work items, update release plan type picklist value, import release plan work items, and remove old picklist values in template. https://github.com/Azure/azure-sdk-tools/issues/8539 https://github.com/Azure/azure-sdk-tools/issues/8541 (La Donna)
- [ ] Send follow-up email based on release scenario - https://github.com/Azure/azure-sdk-tools/issues/9079. 

**Create a release plan workflow** 
![image](https://github.com/user-attachments/assets/bce7338e-ae61-42b2-9b10-8554a3f065c1)

**APEX sub-process workflow** 
![image](https://github.com/user-attachments/assets/2ebcea8e-cbd2-492c-a89f-2ebb074a8712)

**UI changes** 

![Image](https://github.com/user-attachments/assets/8cacbe36-3456-4bfd-928d-ff1bf0f26e0d)

![Image](https://github.com/user-attachments/assets/e38333ac-2055-4f02-8291-5ba4a4495ab3)

**Conditional logic related to UI** 
_Product/APEX Details_ 
1. After product is selected, check the value of the isAPEX field in the epic type = product to determine if the product is in scope for APEX. 
1a. If isAPEX =no, then do not display the questions for isAPEX, isExtension, and the APEX lifecycle milestone drop-down box. Update the isAPEX ADO field in the release plan value to "no".  Proceed to step 5. 
1b.  If isAPEX = yes, then proceed to step 2. 
2. Display the first APEX question "Will this release plan be used to complete requirements for Azure Product Excellence (APEX)?"  yes/no
2a. If answer is "yes" update the isAPEX ADO field in the release plan work item value to be "yes".  Proceed to step 3. 
2b. If answer is "no" update the isAPEX ADO field in the release plan work item value to be "no".  Proceed to step 5. 
3. Is this release plan to complete an APEX exception?  yes/no
3a. If yes, then update the isExtension ADO field in the release plan work item value to "yes".  Proceed to step 4. 
3b. If no, then update the isExtension ADO field in the release plan work item value to "no".  Proceed to step 4. 
4. Display drop-down for user to select "Target APEX lifecycle milestone".  The Release Plan Target Lifecycle ADO field in the release plan work item value is set to the value selected by the user. Proceed to step 5.   ***Note this logic does not permit user to signoff more than one milestone, which is shown in the workflow diagram. Still in discussions with relevant stakeholders as to what scenarios would permit using release plan for signoff of more than one APEX lifecycle milestone. 
5.  User selects release plan type from "Release Plan Type" drop-down box.  

_API Details_ 
1. Update API Spec Type ADO Field in the API Spec work item with the value selected by user. 
2. Update the TargetAPIVersion field in the API Spec work item with the value input by user. 
3. Update the APIPublishDate field in the API Spec work item with the value selected by the user.

**Comment by maririos** [comment]
This issue is to unblock most of the Data plane teams to use the Release Planner

**Comment by maririos** [comment]
Given how the priority right now is to get the Dashboard feature complete, I am moving this issue to backlog

**Comment by ladonnaq** [comment]
@maririos  Once this issue is resolved - https://github.com/Azure/azure-sdk-tools/issues/6638, then hopefully all service partners will be able to find the most closely related product in the drop-down box and create a release plan.  So this requirement should be focused on adding a question to confirm if the work is for an upcoming product launch that is in scope for the APEX launch readiness program or is the work for new functionality that is not in scope for APEX.  I will modify the description and find some time in Jan 2024 to meet with @JonathanCrd to discuss.

**Comment by ladonnaq** [comment]
FYI @maririos - I completed these tasks today. 

1.  Add picklist custom ADO field to the release plan work item. ADO field name = " isAPEX" with string picklist values of "yes" or "no". (La Donna)

2.  Release Plan type picklist values simplified: Private Preview, Public Preview, and GA. Add to existing picklist values, do not remove any values. (La Donna)

**Comment by ladonnaq** [comment]
@ckairen @maririos  I have added screenshots of suggested changes to UI and data/conditional logic associated with the UI changes.

**Comment by maririos** [comment]
@ladonnaq  foolow up questions:
- what are the options for `User selects release plan type from "Release Plan Type" drop-down box.` ? I don't fully understand this question.
- The APEX questions seemd to not belong in the `Product details` section. These seemed to be APEX specific. Maybe we can add another section called `APEX details` below `Product details` and have the questions about isApex and isExtension there
- Trying to see if there is a way to make the question shorter.. will this work? `Will this release plan be used for Azure Product Excellence (APEX)?`

**Comment by ladonnaq** [comment]
> @ladonnaq foolow up questions:
> 
> * what are the options for `User selects release plan type from "Release Plan Type" drop-down box.` ? I don't fully understand this question.
> * The APEX questions seemd to not belong in the `Product details` section. These seemed to be APEX specific. Maybe we can add another section called `APEX details` below `Product details` and have the questions about isApex and isExtension there
> * Trying to see if there is a way to make the question shorter.. will this work? `Will this release plan be used for Azure Product Excellence (APEX)?`

1. The Release Plan Type will be "API Spec - no SDKs", "Beta SDKs", "Stable SDKs".  I have updated the Release Plan Type Field picklist values in ADO. Once engineering has updated the rules/logic, then I will export the Release Plan work items, update the field values, and import. 
2. The APEX question is 100% associated with the product details.  Only products are in scope for APEX.  The question would only be shown if the product is in scope for APEX.  The isAPEX value in the epic-product type is used to determine if the question should be shown. 
3. We could the APEX acronym and then have a tip/help text that explains.  "Will this release plan be used for completing APEX?" Then next question: "Which APEX lifecycle milestone?".
